libparsec/crates/platform_storage/src/workspace.rs

```diff
@@ -217,20 +224,35 @@ impl WorkspaceStorage {
     }
 
     pub async fn set_prevent_sync_pattern(&mut self, pattern: &Regex) -> anyhow::Result<bool> {
-        self.platform.set_prevent_sync_pattern(pattern).await
+        // TODO: Should we skip the next call if the pattern is the same?
+        let applied = self.platform.set_prevent_sync_pattern(pattern).await?;
+
+        self.prevent_sync_pattern = pattern.clone();
+        self.prevent_sync_pattern_applied = applied;
+
+        Ok(applied)
     }
 
-    pub async fn get_prevent_sync_pattern(&mut self) -> anyhow::Result<(Regex, bool)> {
-        self.platform.get_prevent_sync_pattern().await
+    pub fn get_prevent_sync_pattern(&self) -> (&Regex, bool) {
+        (
+            &self.prevent_sync_pattern,
+            self.prevent_sync_pattern_applied,
+        )
     }
 
     pub async fn mark_prevent_sync_pattern_fully_applied(
         &mut self,
         pattern: &Regex,
     ) -> anyhow::Result<bool> {
-        self.platform
+        // TODO: Should we skip the next call if the pattern is different?
```

Comment:
I think it would be good to:
- rename `pattern`  -> `expected_pattern`
- return a dedicated error type:
```rust
enum MarkPreventSyncPatternFullyApplied {
    PatternHasChanged,
    Internal(anyhow::Error)
}
```

This way we can let the caller decide what to do, and on our side we do the comparison between `expected_pattern` and `self.prevent_sync_pattern` as the very first check so that we never modify anything if the pattern is unexpected

Reply:
done